openpype/hosts/maya/plugins/publish/validate_xgen.py

```diff
@@ -0,0 +1,61 @@
+import json
+
+import maya.cmds as cmds
+import xgenm
+
+import pyblish.api
+
+
+class ValidateXgen(pyblish.api.InstancePlugin):
+    """Validate Xgen data."""
+
+    label = "Validate Xgen"
+    order = pyblish.api.ValidatorOrder
+    host = ["maya"]
+    families = ["xgen"]
+
+    def process(self, instance):
+        # Validate only xgen collections are in objectset.
+        nodes = (
+            instance.data["xgenNodes"] +
+            cmds.ls(instance, type="transform", long=True)
+        )
```

Comment:
Lookup `node in nodes` is being doen below. Turns this into a `set` for optimal lookup.

```suggestion
        nodes = set(
            instance.data["xgenNodes"] +
            cmds.ls(instance, type="transform", long=True)
        )
```

Reply:
Sorry, could you elaborate on why a set is needed here?

Reply:
A check whether a string is inside the list is done down below:
```python
        for node in instance:
            if node in nodes:
                continue
```

This will be `On` depending on where it is in the list but the size of the list will have a heavy impact on the performance, whereas lookup in the set is `O1`. 

Some topics:
- https://stackoverflow.com/a/2831242
- https://www.reddit.com/r/Python/comments/j9ncmd/lookup_time_in_a_list_vs_set/
- https://lucasmagnum.medium.com/pythontip-list-vs-set-performance-experiments-dfbe4f72d47f